### PR TITLE
⚡️ Update Image upload service to use Streameth Server

### LINKED ISSUE #822: [TASK] Move all images to the backend
## Description
Right now there is no consistency on the place where images are fetched from, some are fetched from the frontend, some from the backend, which makes debugging and maintaining the code a bit more cumbersome.

## Related User Story
None

## Objectives
- [ ] Analyze current image fetching methods across the application
- [ ] Determine the most efficient and consistent approach for image fetching
- [ ] Implement a unified image fetching strategy from the backend

## Acceptance Criteria
- [ ] All image fetching operations are centralized in either the frontend or backend, not mixed
- [x] Proper tacking of the issue
- [x] Existing image fetching code is refactored to follow the new unified approach

## Additional Information
- Consider implementing a caching strategy to optimize image loading times.
- Ensure that the chosen approach is scalable and can handle potential increases in image volume or traffic.
- Document any API changes or new endpoints created as part of this refactoring process.

### LINKED ISSUE #874: [BUG] Organization banner persists briefly after removal
**Describe the bug**
After removing the organization banner, it continues to be displayed for a few seconds on the organization page instead of immediately showing an empty banner area.

**To Reproduce**
1. Upload a banner to an organization
2. Load the organization page to confirm the banner is visible
3. Remove the banner through the appropriate settings or controls
4. Immediately navigate to or refresh the organization page
5. Observe that the old banner is still visible for a few seconds

**Expected behavior**
Upon removing the banner and viewing the organization page, the banner area should immediately appear empty or display a default state, without showing the previously removed banner.

**Screenshots**

https://github.com/user-attachments/assets/5ca88f54-e999-44f0-a2dc-58b64a7d8b7d

### PR DESCRIPTION
**Pull request by greatsamist**
# Pull Request Info

## Description
Refactored Image Upload service to use streameth api
Fixes #822 
Fixes #874 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: packages/app/app/studio/[organization]/library/[session]/components/ImageDropzone.tsx [hunk 1]
```diff
@@ -39,15 +40,12 @@ const ImageDropzone = forwardRef<HTMLDivElement, ImageDropzoneProps>(
       try {
         const data = new FormData();
         data.set('file', file);
-        data.set('path', path);
-        const res = await fetch('/api/upload', {
```
Comment (xannyxs) [inline packages/app/app/studio/[organization]/library/[session]/components/ImageDropzone.tsx:43]: Please rename this to `imageUrl`

File: packages/app/components/misc/form/imageUpload.tsx [hunk 2]
```diff
@@ -146,19 +147,13 @@ export default function ImageUpload({
           type: file.type,
         })
       );
-      data.set('path', path);
-      const res = await fetch('/api/upload', {
```
Comment (xannyxs) [inline packages/app/components/misc/form/imageUpload.tsx:150]: Please rename the var to imageUrl

File: packages/app/lib/services/imageUploadService.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      body: data,
+    });
+
+    if (!response.ok) {
+      throw 'Error uploading image';
+    }
+
+    return (await response.json()).data;
+  } catch (e) {
+    console.log('error in upload image', e);
```
Comment (xannyxs) [inline packages/app/lib/services/imageUploadService.ts:26]: Bit picky on this, but it is quite a bad practice to catch and then directly throw again. I think it it best to remove the try / catch, and let `imageUploadAction` catch the error.

**Comment by greatsamist** [comment]
> Lastly, can you add the functionality that removes the thumbnail on deletion to avoid unused images?

Nice suggestion! However, this functionality would be better suited for its own pull request. This will help keep changes focused and make it easier to review.

**Comment by greatsamist** [comment]
> > > Lastly, can you add the functionality that removes the thumbnail on deletion to avoid unused images?
> > 
> > 
> > Nice suggestion! However, this functionality would be better suited for its own pull request. This will help keep changes focused and make it easier to review.
> 
> Sure, can you please make an issue of this? I will then merge this PR

New issue created #895

**Review by Mario-SO — APPROVED** [review]

**Pull request merged**